conseil-common/src/main/scala/tech/cryptonomic/conseil/common/rpc/RpcClient.scala

```diff
@@ -0,0 +1,93 @@
+package tech.cryptonomic.conseil.common.rpc
+
+import scala.util.control.NoStackTrace
+
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import org.http4s.{EntityDecoder, EntityEncoder, Header, Method, Uri}
+import org.http4s.client.Client
+import org.http4s.client.dsl.Http4sClientDsl
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient._
+
+// JSON-RPC client according to specification: https://www.jsonrpc.org/specification
+class RpcClient[F[_]: Concurrent](
+    endpoint: String,
+    maxConcurrent: Int,
+    httpClient: Client[F],
+    headers: Header*
+) extends Http4sClientDsl[F]
+    with LazyLogging {
+
+  def stream[P, R](batchSize: Int)(request: Stream[F, RpcRequest[P]])(
+      implicit decode: EntityDecoder[F, Seq[RpcResponse[R]]],
+      encode: EntityEncoder[F, List[RpcRequest[P]]]
+  ): Stream[F, R] =
+    request
+      .chunkN(batchSize)
+      .evalTap { requests =>
+        Concurrent[F].delay(logger.debug(s"Call RPC in a batch of: ${requests.size}"))
+      }
+      .mapAsyncUnordered(maxConcurrent) { chunks =>
+        httpClient
+          .expect[Seq[RpcResponse[R]]](
+            Method.POST(chunks.toList, Uri.unsafeFromString(endpoint), headers: _*)
+          )
+      }
+      .flatMap(Stream.emits)
+      .flatMap {
+        case RpcResponse(_, _, Some(error), _) =>
+          Stream.raiseError[F](error)
```

Comment:
Sorry I didn't spend time on the json-rpc spec.
Since here we conflate both http errors with json rpc ones, is there any guideline on the semantics of such choices?

I mean, should we instead interpret differently the underlying transport protocol failures from the json-rpc failures?
We can always decide to keep it simple for the moment, but I'd prefer this choice to be explicit and documented, in case we plan to take a different path in the future.

Reply:
> Since here we conflate both http errors with json rpc ones

I don't merge errors, the current client implementation can throw `RpcException` (related to spec) or other http4s related exception. In my opinion, it should remain in this way, or I can create the second exception to catch and propagate http4s errors

Reply:
Let's keep it like this for a first shot, and see how it goes